### ISSUE TITLE
OpenComputers support?

### ISSUE THREAD

**Issue by Techokami** [ds84182/CCLights2#16]
It would be nice... :3

**Comment by ds84182** [comment]
The main problem is persistence. I was developing an OC "plugin" for my
newly split core and plugin system. Persisting GPUs would require lots of
work, and would require lots of hard drive space. Each texture, including
monitor textures would have to be saved to a png on UNLOAD, a standard
application using the gpu would require lots of textures. A web
browser-like program I developed for CCLights2 uses a texture for double
buffering, another for the web page, more for images on the page, and
textures for ui elements. An operating system that I scrapped used a screen
sized texture for every application, and each application had a transparent
background with an iOS7-esque blur behind it. More textures were used for
ui elements, another for the home menu background, and yet another texture
was used for double buffering. Now you see why OC support is not feasible.
Another problem is the api. Some functions take advantage of
ComputerCraft's fixed storage. A main function is gpu.import. I depreciated
the ability to load an image from a table full of bytes to introduce a new
feature: image import from file. I don't know how well I can remake
gpu.import in OC. And if I make part of gpu.import lua side, memory usage
comes into play.

To conclude, CCLights2 was based around ComputerCraft, a mod without
limits. A better way to give OC a gpu mod is to probably contact asie and
get him to make a complicated video tape mod. A mod like CCLights2 faces
too many problems in the porting process to OpenComputers. I am not
changing anything with the API just to create a parasitic relationship
between ComputerCraft and OpenComputers.
